### PR TITLE
[WIP][prerelease] Test all supported Ubuntu distros for the specifid ROS distro.

### PR DESCRIPTION
**Pull request by 130s**
This PR is based on #85 so once it's merged this needs rebased to master branch.

There are cases where ROS buildfarm failure occurs on certain Ubuntu distros (e.g. https://github.com/ros-planning/moveit/issues/18#issuecomment-260867116). For more complete coverage the prerelease feature better tests all supported distros (at least Ubuntu) in [REP-0003](http://www.ros.org/reps/rep-0003.html).

### CODE REVIEW

**Comment by 130s** [comment]
On an unrelated note, @ipa-mdl I'd like to invite you to co-maintain this repo. pkgxml is updated accordingly. Thank you for all the great work!

File: industrial_ci/ros_pre-release.sh [hunk 1]
```diff
@@ -43,17 +43,17 @@ CATKIN_DISTRO=$ROS_DISTRO
 
 case "$ROS_DISTRO" in
 "kinetic")
-    os_code_name="xenial"
+    os_code_names=(wily xenial)
     CATKIN_DISTRO="jade"
     ;;
+"indigo"|"jade")
+    os_code_names=(saucy trusty)
+    CATKIN_DISTRO="jade"
```
Comment (ipa-mdl) [inline industrial_ci/ros_pre-release.sh]: CATKIN_DISTRO is just a workaround kinetic, the default ($ROS_DISTRO) is fine for jade and kinetic

File: industrial_ci/ros_pre-release.sh [hunk 2]
```diff
@@ -93,3 +95,16 @@ function run_ros_prerelease() {
 
     return $RESULT_PRERELEASE
 }
+
+function run_ros_prerelease() {
+	retval_prerelease=-1
+	if [ ! "$PRERELEASE_OS_CODENAME" ]; then
```
Comment (ipa-mdl) [inline industrial_ci/ros_pre-release.sh]: PRERELEASE_OS_CODENAME was meant to override the default, e.g. to test another OS variant

**Comment by ipa-mdl** [comment]
>  @ipa-mdl I'd like to invite you to co-maintain this repo.
Thanks :)
Not sure if I can find as much time as needed, but lets try!

**Comment by ipa-mdl** [comment]
I am not sure if we should merge it this way.
It might be more flexible to let the user specify all steps:
```
- ROS_DISTRO=indigo PRERELEASE=true PRERELEASE_OS=ubuntu  PRERELEASE_OS_CODENAME=trusty
- ROS_DISTRO=indigo PRERELEASE=true PRERELEASE_OS=ubuntu  PRERELEASE_OS_CODENAME=saucy
- ROS_DISTRO=kinetic PRERELEASE=true PRERELEASE_OS=debian PRERELEASE_OS_CODENAME=jessie
```
(Alternative syntax: `PRERELEASE_OS="ubuntu trusty"`)

This will result in separate test jobs, so failures can be mapped easily.
Especially the debian tests might be used to identify missing rosdep keys etc.
Per default we could test just the Ubuntu LTS for the given ROS_DISTRO.

**Comment by 130s** [comment]
Probably we should close this in favor of https://github.com/ros-industrial/industrial_ci/pull/174 that is already merged.

**Pull request closed without merging**